src/NuGet.Core/NuGet.Packaging/Signing/Timestamp/Rfc3161TimestampRequestNetstandard21Wrapper.cs

```diff
@@ -0,0 +1,90 @@
+    {
+        private static readonly HttpClient HttpClient = new HttpClient();
+        private readonly System.Security.Cryptography.Pkcs.Rfc3161TimestampRequest _rfc3161TimestampRequest;
+
+        public Rfc3161TimestampRequestNetstandard21Wrapper(
+            byte[] messageHash,
+            HashAlgorithmName hashAlgorithm,
+            Oid requestedPolicyId,
+            byte[] nonce,
+            bool requestSignerCertificates,
+            X509ExtensionCollection extensions)
+        {
+            _rfc3161TimestampRequest = System.Security.Cryptography.Pkcs.Rfc3161TimestampRequest.CreateFromHash(
+                new ReadOnlyMemory<byte>(messageHash),
+                hashAlgorithm,
+                requestedPolicyId,
+                nonce,
+                requestSignerCertificates,
+                extensions);
+        }
+
+        public async Task<IRfc3161TimestampToken> SubmitRequestAsync(Uri timestampUri, TimeSpan timeout)
+        {
+            if (timestampUri == null)
+            {
+                throw new ArgumentNullException(nameof(timestampUri));
+            }
+
+            if (!timestampUri.IsAbsoluteUri)
+            {
+                throw new ArgumentException(
+                    Strings.AnAbsoluteUriIsRequired, nameof(timestampUri));
+            }
+
+            if (timestampUri.Scheme != Uri.UriSchemeHttp && timestampUri.Scheme != Uri.UriSchemeHttps)
+            {
+                throw new ArgumentException(
+                    Strings.HttpOrHttpsIsRequired, nameof(timestampUri));
+            }
+
+            using (var content = new ReadOnlyMemoryContent(_rfc3161TimestampRequest.Encode()))
+            {
+                content.Headers.ContentType = new MediaTypeHeaderValue("application/timestamp-query");
+                using (HttpResponseMessage httpResponse = await HttpClient.PostAsync(timestampUri, content))
+                {
+                    if (!httpResponse.IsSuccessStatusCode)
+                    {
+                        throw new CryptographicException(
+                            string.Format(
+                                CultureInfo.CurrentCulture,
+                                Strings.TimestampServiceRespondedError,
+                                (int)httpResponse.StatusCode,
+                                httpResponse.ReasonPhrase));
+                    }
+
+                    if (!string.Equals(httpResponse.Content.Headers.ContentType.MediaType, "application/timestamp-response", StringComparison.OrdinalIgnoreCase))
+                    {
+                        throw new CryptographicException(Strings.TimestampServiceRespondedInvalidFormat);
+                    }
+
+                    var data = await httpResponse.Content.ReadAsByteArrayAsync();
+
+                    System.Security.Cryptography.Pkcs.Rfc3161TimestampToken response = _rfc3161TimestampRequest.ProcessResponse(data, out int _bytesConsumed);
```

Comment:
You misread [my earlier comment](https://github.com/NuGet/NuGet.Client/pull/3168#discussion_r363852075).

I said:  "You can suppress the message by renaming the local to _, which is the conventional name for an unused out variable."  So, `bytesConsumed` -> `_` not `bytesConsumed` -> `_bytesConsumed`.

Reply:
This feature has a name "discards", which has a [docs page](https://docs.microsoft.com/en-us/dotnet/csharp/discards), which includes a section specifically on [calls to methods with out parameters](https://docs.microsoft.com/en-us/dotnet/csharp/discards#calls-to-methods-with-out-parameters).

Reply:
Thanks for both of you :) Fixed.